ext/mysqlnd/mysqlnd_result.c

```diff
@@ -302,7 +302,7 @@ mysqlnd_query_read_result_set_header(MYSQLND_CONN_DATA * conn, MYSQLND_STMT * s)
 				if (FAIL == (ret = result->m.read_result_metadata(result, conn))) {
 					/* For PS, we leave them in Prepared state */
 					if (!stmt && conn->current_result) {
-						mnd_efree(conn->current_result);
+						result->m.free_result(result, TRUE);
```

Comment:
```suggestion
						result->m.free_result(conn->current_result, TRUE);
```
For consistency.

Reply:
See  https://github.com/php/php-src/pull/15445#discussion_r1720144057.